frontend/src/components/UpcomingReminder/UpcomingReminder.jsx

```diff
@@ -0,0 +1,93 @@
+import './UpcomingReminder.css'
+import {useDispatch} from 'react-redux' 
+import {fetchUserReminders, selectReminders} from '../../store/reminderReducer'
+import { selectPets } from '../../store/petReducer'
+import { selectCurrentUser } from "../../store/sessionReducer"
+import { useSelector } from 'react-redux'
+import { useEffect } from 'react'
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faSyringe, faPills} from "@fortawesome/free-solid-svg-icons"
+import { faCalendarCheck} from "@fortawesome/free-regular-svg-icons"
+
+const UpcomingReminder = () => {
+    const dispatch = useDispatch()
+    const currentUser = useSelector(selectCurrentUser)
+    const reminders = useSelector(selectReminders) //all reminders [{}, {},..]
+    const pets = useSelector(selectPets)
+    const currentUserReminder = reminders.filter(ele  => ele.user === currentUser._id.toString())
+    const today = new Date()
+    const diffDays = []
+    const petIds = []
+    const upcomingReminders = currentUserReminder.filter((ele,idx) => {
+        let due = new Date(currentUserReminder[idx].dueDate)
+        let diff = Math.ceil((due - today) / (1000 * 60 * 60 * 24));
+        if(diff <= 5) {
+            diffDays.push(diff)
+            petIds.push(ele.pet)
+            return ele
+        }
+    })
+    
+    console.log(upcomingReminders)
+    const petNames = []
+    const petImages = []
+
+    for(let i = 0; i< petIds.length; i++) {
+        for(let j = 0; j < pets.length; j++) {
+            if(petIds[i] && petIds[i] === pets[j]._id) {
+                petNames.push(pets[j].name)
+                petImages.push(pets[j].imageUrl)
+                break
+            } else if(petIds[i] === undefined){
+                petNames.push(undefined)
+                petImages.push(undefined)
+                break
+            }
+        }
+    }
+
+    useEffect(() => {
+        dispatch(fetchUserReminders(currentUser._id))
+    }, [dispatch])
```

Comment:
Looks like the linter wants currenUser._id added to the dependency array, but otherwise looks good!